checks/extensions/ocspmuststaple/ocspmuststaple.go

```diff
@@ -0,0 +1,37 @@
+package ocspmuststaple
+
+import (
+	"crypto/x509/pkix"
+	"encoding/asn1"
+
+	"github.com/globalsign/certlint/certdata"
+	"github.com/globalsign/certlint/checks"
+	"github.com/globalsign/certlint/errors"
+)
+
+const checkName = "OCSP Must Staple Extension Check"
+
+// See RFC 7633
+var extensionOid = asn1.ObjectIdentifier{1, 3, 6, 1, 5, 5, 7, 1, 24}
+
+func init() {
+	checks.RegisterExtensionCheck(checkName, extensionOid, nil, Check)
+}
+
+// Check performs a strict verification on the extension according to the standard(s)
+func Check(ex pkix.Extension, d *certdata.Data) *errors.Errors {
+	var e = errors.New(nil)
+
+	// RFC 7633 only defines this extension for PKIX end-entity certificates,
+	// certificate signing requests, and certificate signing certificates (CAs)
+	if d.Type == "OCSP" {
```

Comment:
You might want to check IsCa and or the key usage here, the current check is is only covering an edge case.

Reply:
The check `d.Type == "OCSP"` doesn't check for end-entity certificates, it would only apply for an OCSP signing certificate. I think you are confused with the OCSPNoCheck extension who MUST only be applied on OCSP signing certificates.

Reply:
> The check d.Type == "OCSP" doesn't check for end-entity certificates, it would only apply for an OCSP signing certificate.

Yup. I wasn't trying to explicitly check for end-entity certificates with this clause. That's why the comment says "end-entity certificates, certificate signing requests, and certificate signing certificates".  I was trying to find certificates that were not end-entity, CSRs, or CA certificates. In my scan of `certdata/types.go` it seemed like `OCSP` was the only type that didn't meet those criteria so I checked for it explicitly.

I'm going to dump this check entirely. The `certdata/type.go` package and its types seem too coarse to be especially useful here and your suggestion to check the key usage explicitly is probably better.